packages/shared/lib/services/sync/config/deploy.service.ts

```diff
@@ -835,6 +844,31 @@ async function compileDeployInfo({
     };
 }
 
+async function markSyncConfigAsInactive(oldSyncConfigId: number): Promise<void> {
+    return await db.knex.transaction(async (trx) => {
+        // mark sync config as inactive
+        await trx.from<SyncConfig>(TABLE).update({ active: false }).where({ id: oldSyncConfigId });
+
+        // update sync_config_id in syncs table to point to active sync config
+        await trx.raw(
+            `
+            UPDATE nango._nango_syncs
+            SET sync_config_id = (
+                SELECT active_config.id
+                FROM nango._nango_sync_configs as old_config
+                JOIN nango._nango_sync_configs as active_config
+                ON old_config.sync_name = active_config.sync_name
+                AND old_config.nango_config_id = active_config.nango_config_id
+                AND old_config.environment_id = active_config.environment_id
+                WHERE old_config.id = ?
+                AND active_config.active = true
+            )
+            WHERE sync_config_id = ?`,
+            [oldSyncConfigId, oldSyncConfigId]
+        );
```

Comment:
not a huge fan of a complex query like this but to be able to match a old sync config with the new added one it would require a heavy refactor of `deploy` and `deployPreBuilt` functions, which I didn't want to do right now